cmd/argoexec/commands/wait.go

```diff
@@ -32,20 +30,13 @@ func waitContainer(ctx context.Context) error {
 	defer stats.LogStats()
 	stats.StartStatsTicker(5 * time.Minute)
 
-	// use a function to constrain the scope of ctx
```

Comment:
Do we consider using the `cmd.Context()` as the root-context ?

Reply:
I've been working with this, and I'm not sure my PR will fix the issues.

Reply:
Hope my work can help you 😄

https://github.com/argoproj/argo-workflows/issues/10490#issuecomment-1427294122